dash/backend/src/modules/falco/service/falco.service.ts

```diff
@@ -120,6 +125,126 @@ export class FalcoService {
+                falcoLogTags: instanceToPlain(newFalcoLog[0].raw["tags"]),
+                moreDetailsLink: `https://dev-m9sweeper.intelletive.com/private/clusters/${clusterId}/falco/more/${falcoId}/signature/${falcoSignature}`,
+            }
+        }).catch(e => {
+            console.log('Error sending falco email: ' + e);
+        });
+        return savedTime;
+    }
+
+    async addFalcoEmail(emailSentTime: number, clusterId: number, falcoSignature: string): Promise<any>{
+        return this.falcoDao.addFalcoEmail(emailSentTime, clusterId, falcoSignature);
+    }
+
+    async falcoEmailAlreadySent(clusterId: number, falcoSignature: string): Promise<any>{
+        return this.falcoDao.falcoEmailAlreadySent( clusterId, falcoSignature);
+    }
+
+    async sendFalcoEmailAlert(clusterId: number, newFalcoLog: FalcoDto) {
+        const falcoSetting = await this.findFalcoSettingByClusterId(clusterId);
+
+        if (!falcoSetting) {
+            return; // no settings means there's no email to send!
+        }
+
+        // get all admin email addresses
+        const allAdminEmail = await this.getAllAdminsToMail();
+        // To test email: const allAdminEmailArray = ['some_email_address'] and comment out forEach();
+        const allAdminEmailArray = [''];
+        allAdminEmail.forEach( element => allAdminEmailArray.push(element.email));
+
+        // Parse data from new falco log fields
+        const falcoId = newFalcoLog[0].id;
+        const falcoSignature = newFalcoLog[0].anomaly_signature;
+        const falcoNamespace = newFalcoLog[0].namespace;
+        const falcoSeverity = newFalcoLog[0].level;
+
+        // Parse data from falco settings json fields
+        const severityLevel = falcoSetting["severity_level"];
+        const anomalyFrequency = (falcoSetting['anomaly_notification_frequency'] * 24 * 60 * 60 * 1000);
+        const weekDay = falcoSetting['weekday'];
+        const emailList = falcoSetting['email_list'].split(',');
+        const cleanEmailList = [];
+        for (const ele of emailList) {
+            cleanEmailList.push(ele.trim());
+        }
+        const whoToNotify = falcoSetting['who_to_notify'];
+
+
+        if (!severityLevel.includes(falcoSeverity)) {
+            return; // no need to send email - does not match the severity level we send emails for
+        }
+
+        // Have we already sent an email ?
+        const lastEmailSentTime = await this.falcoEmailAlreadySent(clusterId, falcoSignature);
+
+        // if no email record then send an email now
+        if (lastEmailSentTime === null || lastEmailSentTime <= (Date.now() - lastEmailSentTime)) {
+            await this.sendFalcoEmailAndAddFalcoEmailRecord(whoToNotify, allAdminEmailArray, emailList, clusterId, falcoId, falcoSeverity, falcoNamespace, falcoSignature, newFalcoLog);
+        } else {
+            // when was the last email sent?
+            const timeDifference = Date.now() - lastEmailSentTime;
```

Comment:
I don't believe you need to check the timeDifference here because you are already checking it above.